src/fmu/dataio/_metadata.py

```diff
@@ -50,25 +51,24 @@ def default_meta_dollars() -> dict:
 
 def generate_meta_tracklog() -> list[dict]:
     """Initialize the tracklog with the 'created' event only."""
-
-    dtime = datetime.datetime.now(timezone.utc).isoformat()
-    user = getpass.getuser()
-    sysinfo = {
-        "fmu-dataio": {
-            "version": dataio.__version__,
-        },
-    }
-    _kmd = read_named_envvar("KOMODO_RELEASE")
-    if _kmd is not None:
-        sysinfo["komodo"] = {"version": _kmd}
-
     return [
-        {
-            "datetime": dtime,
-            "user": {"id": user},
-            "event": "created",
-            "sysinfo": sysinfo,
-        }
+        meta.TracklogEvent(
+            datetime=datetime.datetime.now(timezone.utc),
+            event="created",
+            user=meta.User(id=getpass.getuser()),
+            sysinfo=meta.SystemInformation(
+                fmudataio=meta.VersionInformation(
+                    version=dataio.__version__,
+                ),
+                komodo=meta.VersionInformation(version=kr)
+                if (kr := read_named_envvar("KOMODO_RELEASE"))
+                else None,
+            ),
+        ).model_dump(
+            mode="json",
+            exclude_none=True,
+            by_alias=True,
+        )
```

Comment:
Ideally i do not want todo this serialization, but to mutch work to get around it atm. Once the pydantic coverage is good, this can be removed.

Reply:
Side note, once we got more and more of these converted to pydantic the call to model_dump should be removed over time.